tests/src/events-provider.consumer.ts

```diff
@@ -4,9 +4,11 @@ import { OnEvent } from '../../lib';
 @Injectable()
 export class EventsProviderConsumer {
   public eventPayload = {};
+  public eventReceiveCount = 0;
```

Comment:
do we need this field?

Reply:
strictly no.
It was how I was able to test that the event was received twice before my change and only once after my change.

If I were to take this out, I would have no test updates for this fix unless you have some other ideas?

Reply:
I guess we could test if `onTestEvent` has been called only once somehow

Reply:
So i tried changing it to a spy
```js
const eventSpy = jest.spyOn(eventsConsumerRef, 'onTestEvent');
expect(eventSpy).toBeCalledTimes(1);
```
that part works, but then the payload fails to validate because the spy got in the way.

Jest documentation says spyOn still calls the original function but that does not seem to be happening? I'll admit i am weaker when it comes to jest and mocking.

Reply:
write another test instead of reusing the `should emit a "test-event" event to providers` one

Reply:
So i have played with this a bit.
Using jest.spyOn at all breaks the OnEvent decorator since the spy removes metadata. Which rules out using the toBeCalledTimes() approach as is.

I could modify the EventsProviderConsumer class (or do it in a new provider) to have a setter for eventPayload vs directly changing it. With that I would be able to mock that set function and count the times it was called. 


Let me know which approach you would like. Adding this eventReceiveCount like I have done already is the smallest change to achieve the goal.

Reply:
I believe we just need to add:

```ts
afterEach(() => {
  // restore the spy created with spyOn
  jest.restoreAllMocks();
});
```

Reply:
Sorry, i was a bit ambiguous about that part.
Even when running my new test alone (it.only), by creating the mock, it effectively deletes the OnEvent decorator (because the mock does not copy the Reflect metadata that the decorator adds) so no events are ever received during that specific test. So as far as I can tell i cannot spy on the function that actually handles the event.

So what I can do is create a provider that looks like:
```ts
  private _eventPayload = {};
  set eventPayload(value) {
    this._eventPayload = value;
  }

  get eventPayload() {
    return this._eventPayload;
  }

  @OnEvent('test.*')
  onTestEvent(payload: Record<string, any>) {
    this.eventPayload = payload;
  }
```
which would then allow me to mock the setter with
```js
const eventSpy = jest.spyOn(eventsConsumerRef, 'eventPayload', 'set');
expect(eventSpy).toBeCalledTimes(1);
```

Just looking for confirmation you still would prefer this approach vs the simple counter

Reply:
got it.

I don't know, I feel that might be a better way to check if `onTestEvent` has been called. But I didn't have time to think on that now.

I dislike the ideia of adding a field on that class just to make the test easier to write.

Reply:
Ok refactored test as above and removed the counter approach.